src/query/frontend/ast/cypher_main_visitor.cpp

```diff
@@ -1926,16 +1926,21 @@ antlrcpp::Any CypherMainVisitor::visitNodePattern(MemgraphCypher::NodePatternCon
 }
 
 antlrcpp::Any CypherMainVisitor::visitNodeLabels(MemgraphCypher::NodeLabelsContext *ctx) {
-  std::vector<LabelIx> labels;
+  std::vector<std::variant<LabelIx, Expression *>> labels;
   for (auto *node_label : ctx->nodeLabel()) {
-    if (node_label->labelName()->symbolicName()) {
-      labels.emplace_back(AddLabel(std::any_cast<std::string>(node_label->accept(this))));
+    if (node_label->labelName()) {
+      if (node_label->labelName()->symbolicName()) {
+        labels.emplace_back(AddLabel(std::any_cast<std::string>(node_label->accept(this))));
+      } else {
+        // If we have a parameter, we have to resolve it.
+        const auto *param_lookup = std::any_cast<ParameterLookup *>(node_label->accept(this));
+        const auto label_name = parameters_->AtTokenPosition(param_lookup->token_position_).ValueString();
+        labels.emplace_back(storage_->GetLabelIx(label_name));
```

Comment:
It shouldn't. It will check in storage if label with given name already exists, if it does it will return it's LabelIx, otherwise it will add LabelIx for given name.